DQMOffline/Trigger/plugins/MuonMonitor.cc

```diff
@@ -0,0 +1,288 @@
+  me.denominator = ibooker.book1D(histname+"_denominator", histtitle+" (denominator)", nbins, arr);
+}
+void MuonMonitor::bookME(DQMStore::IBooker &ibooker, MuonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, double ymin, double ymax)
+{
+  me.numerator   = ibooker.bookProfile(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, ymin, ymax);
+  me.denominator = ibooker.bookProfile(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, ymin, ymax);
+}
+void MuonMonitor::bookME(DQMStore::IBooker &ibooker, MuonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax)
+{
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, nbinsY, ymin, ymax);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, nbinsY, ymin, ymax);
+}
+void MuonMonitor::bookME(DQMStore::IBooker &ibooker, MuonME& me, const std::string& histname, const std::string& histtitle, const std::vector<double>& binningX, const std::vector<double>& binningY)
+{
+  int nbinsX = binningX.size()-1;
+  std::vector<float> fbinningX(binningX.begin(),binningX.end());
+  float* arrX = &fbinningX[0];
+  int nbinsY = binningY.size()-1;
+  std::vector<float> fbinningY(binningY.begin(),binningY.end());
+  float* arrY = &fbinningY[0];
+
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, arrX, nbinsY, arrY);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, arrX, nbinsY, arrY);
+}
+
+void MuonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+
+  histname = "muon_pt"; histtitle = "muon PT";
+  bookME(ibooker,muonME_,histname,histtitle,muon_binning_.nbins,muon_binning_.xmin, muon_binning_.xmax);
+  setTitle(muonME_,"Muon pT [GeV]","events / [GeV]");
+
+  histname = "muon_pt_variable"; histtitle = "muon PT";
+  bookME(ibooker,muonME_variableBinning_,histname,histtitle,muon_variable_binning_);
+  setTitle(muonME_variableBinning_,"Muon pT [GeV]","events / [GeV]");
+
+  histname = "muonVsLS"; histtitle = "muon pt vs LS";
+  bookME(ibooker,muonVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,muon_binning_.xmin, muon_binning_.xmax);
+  setTitle(muonVsLS_,"LS","Muon pT [GeV]");
+
+  histname = "muon_phi"; histtitle = "Muon phi";
+  bookME(ibooker,muonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+  setTitle(muonPhiME_,"Muon #phi","events / 0.1 rad");
+
+
+  histname = "muon_eta"; histtitle = "Muon eta";
```

Comment:
isn't this analysis interested in di-muon mass as well ?
why don't you add it ?